Sources/Amplitude/Plugins/iOS/iOSUserInteractionCapture/CustomUIGestureRecognizers.swift

```diff
@@ -0,0 +1,312 @@
+
+    /// The minimum press duration to detect a long press.
+    private static let minimumPressDuration: TimeInterval = 0.5
+
+    /// The number of clicks required to classify the followed clicks as rage clicks during
+    /// the `rageClickTimeWindow` period.
+    private static let rageClickCountThreshold = 5
+
+    /// The time window to detect rage clicks.
+    private static let rageClickTimeWindow: TimeInterval = 0.8
+
+    /// The delegate that manages user interactions options.
+    private var captureDelegateHandler: () -> UserInteractionCaptureDelegate?
+
+    /// The target element when the screen was first touched.
+    private var initialTarget: AccessibilityTarget?
+
+    /// The root view of the screen when it was first touched.
+    private var initialRootView: UIView?
+
+    /// The initial point where the scroll gesture began.
+    private var initialTouchLocation: CGPoint?
+
+    /// The start time of the press gesture.
+    private var startTime: TimeInterval?
+
+    /// Specifies whether the followed clicks are rage clicks.
+    private var isRageClicking = false
+
+    /// The element targetted by a series of rage clicks.
+    private var currentRageClickTarget: NSObject?
+
+    /// Recent taps and the corresponding time associated with the tap.
+    private var recentTaps: [(target: AccessibilityTarget, time: TimeInterval)] = []
+
+    // MARK: - Life Cycle
+
+    init(for captureDelegateHandler: @escaping () -> UserInteractionCaptureDelegate?) {
+        self.captureDelegateHandler = captureDelegateHandler
+        super.init(target: nil, action: nil)
+    }
+
+    // MARK: - Overridden Methods
+
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent) {
+        super.touchesBegan(touches, with: event)
+
+        guard
+            let captureDelegate = captureDelegateHandler()
+        else {
+            return
+        }
+
+        initialTouchLocation = touches.first?.location(in: captureDelegate.keyWindow)
+        initialRootView = captureDelegate.keyWindow
+        startTime = Date().timeIntervalSince1970
+
+        if let initialTouchLocation,
+           let target = captureDelegate.accessibilityTargets.first(where: {
+            $0.shape.contains(initialTouchLocation) &&
+            ($0.type.contains(.button) || $0.type.contains(.link))
+        }) {
+            initialTarget = target
+        }
+    }
+
+    override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent) {
+        super.touchesEnded(touches, with: event)
+
+        guard
+            let captureDelegate = captureDelegateHandler()
+        else {
+            return
+        }
+
+        defer {
+            self.initialTarget = nil
+        }
+
+        guard
+            let initialTouchLocation,
+            let startTime,
+            let initialTarget,
+            let initialRootView,
+            let endTouchLocation = touches.first?.location(in: initialRootView),
```

Comment:
What happens if there are multiple overlapping a11y elements? how would the system resolve this?

Reply:
The first element from the hierarchy parser is the top most element.